src/main/java/gift/service/ProductService.java

```diff
@@ -24,4 +33,25 @@ public void validateProductException(Product product) {
             throw new IllegalArgumentException("상품 이름에 '카카오'는 포함할 수 없습니다.");
         }
     }
+
+    public Product save(Product product) {
+        return productRepository.save(product);
```

Comment:
```suggestion
    @Transactional
    public Product save(Product product) {
        return productRepository.save(product);
```

- 트랜잭션 단위를 묶어주는게 어떨까요?

Reply:
확인하였습니다. 트랜젝션에 대해서 조금 조사해 보았는데 트랜젝션은 상태를 성공, 실패로만 다루어서 하나만 가능하도록 하는 기능이고, save에서 사용하는 이유는 저장이 나중에 한번에 이루어지지 않으면 오류가 생길 수 있어 미리 확장성을 위해 묶는 것인지 궁금합니다.

Reply:
- save가 하나의 트랜잭션 단위일 수도 있고, 이야기주신 것처럼 나중에 묶일 수도 있어요. 다만 미리 확장성을 위해 트랜잭션을 설정하는건 아니에요. 언젠가는 Spring AOP를 학습하며 Transcational 동작방식을 학습하게 되긴 할텐데요. 모든 메서드에 붙여줘야 하는건 아닙니다. 저건 그저 예시일 뿐이구요. 오히려 지금 생각해보셔야 할 건, Service Layer의 역할은 무엇인가 예요. 그리고 거기서 많이들 얘기하는 것 중 하나가, Transaction의 단위라는 표현이 있어요. 어떤 의미인지 좀 더 학습하고 고민해보시길 바래요. 그래요 Service Layer Test를 작성할 때, 어떤 목적으로 어떤 테스트를 작성해야 할지도 가늠할 수 있을거라서요.